src/codegen.cpp

```diff
@@ -1324,88 +1322,78 @@ void *jl_get_llvmf(jl_function_t *f, jl_tupletype_t *tt, bool getwrapper, bool g
         linfo = jl_get_unspecialized(linfo);
     }
 
-#if defined(USE_ORCJIT) || defined(USE_MCJIT)
-    if (linfo->functionObjects.functionObject == NULL && linfo->functionObjects.specFunctionObject == NULL) {
-        jl_compile_linfo(linfo, NULL);
-    }
-    if (getdeclarations) {
-        JL_GC_POP();
-        if (getwrapper || linfo->functionObjects.specFunctionObject == NULL)
-            return linfo->functionObjects.functionObject;
+    if (!getdeclarations) {
+        Function *llvmDecl = nullptr;
```

Comment:
the latter implies the former, but not the other way around. with gcc 4 and old llvm's llvm-config flags we can't assume c++11